extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -14,6 +19,58 @@ export class AmpStoryShoppingAttachment extends AmpStoryPageAttachment {
 
     /** @private {?Element} */
     this.container_ = null;
+
+    /** @private @const {?../../amp-story/1.0/amp-story-store-service.AmpStoryStoreService} */
+    this.storeService_ = null;
+
+    /** @private {?../../../src/service/localization.LocalizationService} */
+    this.localizationService_ = null;
+  }
+
+  /**
+   * Updates the CTA based on the shopping data.
+   * @param {string} pageIndex
+   * @param {string} pageId
+   * @private
+   */
+  updateCta_(pageIndex, pageId) {
+    const shoppingTag = this.element.ownerDocument
+      .getElementById(pageId)
+      .getElementsByTagName('amp-story-shopping-tag');
+    const numShoppingTags = shoppingTag != null ? shoppingTag.length : 0;
+
+    const pageOutlink = document.getElementsByClassName(
+      'i-amphtml-story-page-open-attachment-host'
+    )[pageIndex].shadowRoot.children[1];
```

Comment:
You already have the page ID. Why not use that to get a reference to the page instead of index?

Reply:
If we change the dom structure, this will break.
Do not access by index, access by querySelector.

Accessing by querySelector is also easier to read since we name our elements descriptively.

Reply:
Do not access elements via index. Use a querySelector. 
This is fragile. If the dom structure changes this will break. 

Using class selectors makes the codebase easier to use, for instance we can search `i-amphtml-story-page-open-attachment` and see where it's being modified. 

Please ping me if my comments don't make sense, I'm happy to elaborate!

Reply:
It was already updated to use queryselector, I updated the code in [yesterday's commit](https://github.com/ampproject/amphtml/pull/37008/commits/8d673714ae1c654d3488fc6e57109fc8566a0298) - please carefully look through the latest changes (not the review comments as they may point to outdated code such as this one) before commenting. Thanks.

Reply:
There are still index selectors [here](https://github.com/ampproject/amphtml/blob/ca444fd87d244a20700b78828a10e58ef33d64b7/extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js#L86) and [here](https://github.com/ampproject/amphtml/blob/ca444fd87d244a20700b78828a10e58ef33d64b7/extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js#L63).
Please update these.

Please be sure to apply review comments like this to the entirety of the document!